src/Linters/NoBasicAssignmentFunctionParameterLinter.php

```diff
@@ -0,0 +1,115 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST\Linters;
+
+use type Facebook\HHAST\{
+  EditableNode,
+  EditableList,
+  FunctionCallExpression,
+  BinaryExpression,
+  EqualToken,
+  DelimitedComment,
+  ListItem,
+  CommaToken,
+  WhiteSpace
+};
+
+use function Facebook\HHAST\;
+use namespace Facebook\TypeAssert;
+use namespace HH\Lib\{C, Str, Vec};
+
+class NoBasicAssignmentFunctionParameterLinter
+  extends AutoFixingASTLinter<FunctionCallExpression> {
+
+    <<__Override>>
+    protected static function getTargetType(): classname<FunctionCallExpression> {
+      return FunctionCallExpression::class;
+    }
+
+    <<__Override>>
+    public function getLintErrorForNode(
+      FunctionCallExpression $node,
+      vec<EditableNode> $parents,
+    ): ?FixableASTLintError<FunctionCallExpression> {
+      $args = $node->getArgumentList();
+      if ($args === null) {
+        return null;
+      }
+      $exps = $args->getItemsOfType(BinaryExpression::class);
+      if ($exps === null) {
+        return null;
+      }
+      $assignment_exps = vec[];
+      foreach ($exps as $exp) {
+        if ($exp instanceof BinaryExpression && $exp->getOperator() instanceof EqualToken) {
+          $assignment_exps[] = $exp;
+        }
+      }
+      if (C\is_empty($assignment_exps)) {
+        return null;
+      }
+      $original = $node->getCode();
+      $fixed = $this->getFixedNode($node)->getCode();
+      return new FixableASTLintError(
+        $this,
+        "Basic assignment is not allowed in ".$original."; consider changing to ".$fixed,
```

Comment:
can you comment with an example of the CLI output here?

The $original and $fixed feels redundant, giving that FixableLintErrors should always output them.

Reply:
if the issue ise that it's not getting the position correct because of the new EditableList, could blame to the whole Binary Expression

Reply:
Basic assignment is not allowed in 
foo(1, $b = 2); consider changing to 
foo(1, /* $b = */ 2)
  Linter: Facebook\HHAST\Linters\NoBasicAssignmentFunctionParameterLinter
  Location: ../Hack/fun.php:7:0
  Code:
  -
  -foo(1, $b = 2)
  Suggested fix:
  +
  +foo(1, /* $b = */ 2)

Reply:
Not sure which position to blame in the case where there are multiple assignments.

i.e. foo(1, $a = 2, 3, 4, $b = 5)

Reply:
Blame to the whole function call expression?